LandSandBoat/server
# Blu magic core updates

**_I affirm:_**

## What does this pull request do?

This ports the SATA code from CEXI for blu magic.

It also enables TA to function for blu magic skillchain damage

End result is (using cannonball as an example)
- a SA cannonball will auto crit
- if taChar is in position
  - a TA cannonball will transfer enmity
  - a TA cannonball that skillchains will transfer enmity of both spell and skillchain

`PHYSICAL_POTENCY` now gives acc and bonus to cratio

Finally, some adjustments were made to support `params.bonusacc` and `params.critchance`. If these params aren't passed both will be zero by default

_Also_ (sorry for the late discovery, but didn't want to make a new PR since this one is still open): Found that the base multiplier is being set to 1 unless Chain Affinity is in effect. This is wrong: https://www.bg-wiki.com/ffxi/Calculating_Blue_Magic_Damage

The base multiplier is in the table as ftp0 and is only applied to the first hit.

## Steps to test these changes

`!stun` and `!immortal` a mob
position for SA and compare damage before and after cannonball
position for TA and remove `!stun` after to see the mob target the player
TA for skillchain damage is a bit trickier to test, but you can use `!getenmity` to confirm your enmity doesn't spike after using TA (open the skillchain with the second player)
